### PR TITLE
update Sentry

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR updates Sentry client to v10 by following these guides:

- https://hexdocs.pm/sentry/upgrade-9-x.html
- https://hexdocs.pm/sentry/upgrade-10-x.html

If we update Oban from 2.17.2 to >2.17.6, we can also start using https://hexdocs.pm/sentry/oban-integration.html#cron-support

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -47,8 +47,7 @@ RUN npm run deploy --prefix ./tracker && \
   mix assets.deploy && \
   mix phx.digest priv/static && \
   mix download_country_database && \
-  # https://hexdocs.pm/sentry/Sentry.Sources.html#module-source-code-storage
-  mix sentry_recompile
+  mix sentry.package_source_code
```
Comment (ruslandoga) [inline Dockerfile:50]: https://hexdocs.pm/sentry/upgrade-10-x.html#actively-package-your-source-code

File: config/runtime.exs [hunk 2]
```diff
@@ -339,21 +339,17 @@ else
     database: get_var_from_path_or_env(config_dir, "DATABASE_NAME", "plausible")
 end
 
-included_environments = if sentry_dsn, do: ["prod", "staging", "dev"], else: []
 sentry_app_version = runtime_metadata[:version] || app_version
 
 config :sentry,
   dsn: sentry_dsn,
   environment_name: env,
-  included_environments: included_environments,
```
Comment (ruslandoga) [inline config/runtime.exs:348]: https://hexdocs.pm/sentry/upgrade-10-x.html#stop-using-included_environments

File: config/runtime.exs [hunk 3]
```diff
@@ -339,21 +339,17 @@ else
...
   release: sentry_app_version,
   tags: %{app_version: sentry_app_version},
-  enable_source_code_context: true,
-  root_source_code_path: [File.cwd!()],
   client: Plausible.Sentry.Client,
   send_max_attempts: 1,
   filter: Plausible.SentryFilter,
-  before_send_event: {Plausible.SentryFilter, :before_send}
+  before_send: {Plausible.SentryFilter, :before_send}
```
Comment (ruslandoga) [inline config/runtime.exs:352]: https://hexdocs.pm/sentry/upgrade-10-x.html#rename-before_send_event-to-before_send

File: lib/sentry_filter.ex [hunk 4]
```diff
@@ -4,7 +4,6 @@ defmodule Plausible.SentryFilter do
   """
   @behaviour Sentry.EventFilter
 
-  def exclude_exception?(%Sentry.CrashError{}, _source), do: true
```
Comment (ruslandoga) [inline lib/sentry_filter.ex:7]: https://github.com/getsentry/sentry-elixir/blob/master/CHANGELOG.md#breaking-changes

**Comment by ruslandoga** [comment]
Tests fail due to https://github.com/getsentry/sentry-elixir/issues/663, the fix for which hasn't been released yet https://github.com/getsentry/sentry-elixir/issues/682

File: config/runtime.exs [hunk 5]
```diff
@@ -339,21 +339,16 @@ else
...
   dsn: sentry_dsn,
   environment_name: env,
-  included_environments: included_environments,
   release: sentry_app_version,
   tags: %{app_version: sentry_app_version},
-  enable_source_code_context: true,
-  root_source_code_path: [File.cwd!()],
   client: Plausible.Sentry.Client,
   send_max_attempts: 1,
-  filter: Plausible.SentryFilter,
```
Comment (ruslandoga) [inline config/runtime.exs:355]: https://hexdocs.pm/sentry/Sentry.EventFilter.html is deprecated

File: lib/plausible/sentry/client.ex [hunk 6]
```diff
@@ -5,17 +5,6 @@ defmodule Plausible.Sentry.Client do
...
-  @doc """
-  The Sentry.HTTPClient behaviour requires a child spec to be supplied.
-  In this case we don't want Sentry to manage our Finch instances, hence it's fed
-  with a dummy module for the sake of the contract.
-
-  XXX: Submit a Sentry PR making the child spec callback optional.
-  """
-  def child_spec do
-    Task.child_spec(fn -> :noop end)
-  end
```
Comment (ruslandoga) [inline lib/plausible/sentry/client.ex:17]: Is it optional now https://hexdocs.pm/sentry/Sentry.HTTPClient.html#module-child-spec

**Comment by ruslandoga** [comment]
If we update Oban from 2.17.2 to > 2.17.6, we can also start using https://hexdocs.pm/sentry/oban-integration.html#cron-support

File: lib/plausible/application.ex [hunk 7]
```diff
@@ -119,10 +119,10 @@ defmodule Plausible.Application do
 
   defp maybe_add_sentry_pool(pool_config) do
     case Sentry.Config.dsn() do
-      dsn when is_binary(dsn) ->
-        Map.put(pool_config, dsn, size: 50)
+      {"http" <> _rest = url, _, _} ->
```
Comment (ruslandoga) [inline lib/plausible/application.ex:122]:
```elixir
# on my instance
iex(plausible@6266675d6904)1> Sentry.Config.dsn
{"https://o1012425.ingest.sentry.io/api/6643873/envelope/",
 "7f16d5d6ee70465789e0**********", nil}
```

File: lib/plausible_web/live/sentry_context.ex [hunk 8]
```diff
@@ -26,15 +26,9 @@ defmodule PlausibleWeb.Live.SentryContext do
...
-      socket_host =
-        case socket.host_uri do
-          :not_mounted_at_router -> :not_mounted_at_router
-          %URI{host: host} -> host
-        end
-
       request_context =
         %{
-          host: socket_host,
+          url: socket.host_uri,
```
Comment (ruslandoga) [inline lib/plausible_web/live/sentry_context.ex:31]: https://hexdocs.pm/sentry/Sentry.Context.html#t:request_context/0

File: lib/plausible/application.ex (hunk 7 above)
Comment (ruslandoga) [reply lib/plausible/application.ex:122]: And it seems to work fine: https://eh-uo.sentry.io/share/issue/b48495ef6e184186a406cc4b7109e4ea/

**Comment by aerosol** [comment]
Will this still work with Sentry 21.3.0.dev0?

**Comment by ruslandoga** [comment]
Since https://develop.sentry.dev/sdk/envelopes/ are still being used, I'd expect it to work. I'll try it out against 21.3.0.dev0 later today.

**Comment by ruslandoga** [comment]
I failed to get self-hosted sentry running at tag `21.3.0` ... I still think it should work as most of [the changes](https://github.com/getsentry/sentry-elixir/blob/master/CHANGELOG.md) since v8 were in the lib's own API.

**Review by aerosol — APPROVED** [review]

**Pull request merged**